src/main/scala/mesosphere/marathon/core/flow/impl/ReviveOffersActor.scala

```diff
@@ -86,27 +96,40 @@ private class ReviveOffersActor(
     case OffersWanted(true) =>
       log.info("Received offers WANTED notification")
       offersCurrentlyWanted = true
-      reviveOffers()
+      initiateNewSeriesOfRevivesAsLongAsOffersAreWanted()
 
     case OffersWanted(false) =>
-      log.info("Received offers NOT WANTED notification, canceling any TimedChecks")
+      log.info("Received offers NOT WANTED notification, canceling {} revives", revivesNeeded)
       offersCurrentlyWanted = false
+      revivesNeeded = 0
       nextReviveCancellableOpt.foreach(_.cancel())
       nextReviveCancellableOpt = None
   }
 
+  def initiateNewSeriesOfRevivesAsLongAsOffersAreWanted(): Unit = {
```

Comment:
Nit: whether offers are wanted or not is evaluated _before_ calling the function – so the function could simply be called `initiateNewSeriesOfRevives`, as it is initiating and (re-)setting the `revivesNeeded`